Src/PCompiler/CompilerCore/Backend/CSharp/CompilationContext.cs

```diff
@@ -11,7 +12,7 @@ public CompilationContext(ICompilerConfiguration job)
         {
             Names = new CSharpNameManager("PGEN_");
 
-            FileName = $"{ProjectName}.cs";
+            FileName = Path.Combine(job.OutputDirectory.FullName, $"{ProjectName}.cs");
```

Comment:
Instead of this, can you keep the file name as it and then change the writer to write output to output directory.

Reply:
Resolved in 088a3fad5d48b6b694a515ee95746d274f2e52a1